src/jit/codegencommon.cpp

```diff
@@ -7446,7 +7446,7 @@ void CodeGen::genProfilingEnterCallback(regNumber initReg, bool* pInitRegZeroed)
 
             var_types storeType = varDsc->lvaArgType();
             regNumber argReg    = varDsc->lvArgReg;
-            getEmitter()->emitIns_S_R(ins_Store(storeType), emitTypeSize(storeType), argReg, varNum, 0);
+            getEmitter()->emitIns_S_R(ins_Store(storeType, false, argReg), emitTypeSize(storeType), argReg, varNum, 0);
         }
```

Comment:
It seems like instead of changing the signature of the ins_Store/ins_Load functions, you could do (essentially) what genFnPrologCalleeRegArgs() does, and add this code here (and similar code for the load case):
```
        storeType = varTypeIsStruct(storeType) ? TYPE_I_IMPL : storeType;
```